btree/btree.c

```diff
@@ -2664,8 +2650,10 @@ static int64_t btree_put_alloc_phase(struct m0_btree_op *bop)
 			} else {
 				node_op_fini(&oi->i_nop);
 				oi->i_used = bop->bo_arbor->t_height - 1;
+				if (bop->bo_flags & BOF_LOCKALL)
+					lock_op_unlock(tree);
```

Comment:
If this unlocking is done just for calling level_cleanup then as I mentioned above make sure to just pass the current lock state to level_cleanup instead of unlocking here.

Reply:
Need to see if we really want to add lock_acquired parameter for level_cleanup

Reply:
Please discuss this with Yash and Nikhil before arriving at any conclusion.

Reply:
The purpose of unlocking at this stage is not to unlock because we are calling level_calling after that. But to make sure unlock the lock before new traversal. because thread will request for lock in LOCKALL

Reply:
I see that the check and unlock was added in this PR so it makes me assume that this unlock is happening to create the right environment before calling level_cleanup

Reply:
okay but this is not the case we are unlocking whenever we want to start traversing again or we are done with operation . and in both the scenarios we are calling  level_cleanup. thats why we are unlocking so indirectly level_cleanup gets called in unlock mode.